scripts/rest_client.py

```diff
@@ -174,6 +174,13 @@ def show_config(self):
         """
         return self._get("show_config")
 
+    def ping(self):
+        """simple ping"""
+        if self.show_config()['state'] == "running":
```

Comment:
Don't worry about tests - the script is not covered by tests.

---

I'm not sure if using `show_config` is the best option ...

there is a specific "ping" rest method which requires no authentication ... so it might be better to use that instead.

---

either way - this fails with `TypeError: 'NoneType' object is not subscriptable`  in case of a connection error...

```
 python scripts/rest_client.py --config ../xmatt/config.json ping
2021-03-02 10:47:19,688 - ft_rest_client - WARNING - Connection error
Traceback (most recent call last):
  File "scripts/rest_client.py", line 394, in <module>
    main(args)
  File "scripts/rest_client.py", line 389, in main
    print(json.dumps(getattr(client, command)(*args["command_arguments"])))
  File "scripts/rest_client.py", line 179, in ping
    if self.show_config()['state'] == "running":
TypeError: 'NoneType' object is not subscriptable

```

Reply:
I changed the code, incase of connection error returns `{"status": "not_running"}`

Reply:
I used show_config endpoint because in the original issue that was the recommended endpoint.